File: trunk/quote/Projects/PluginModel/quote/Detail.vb
Imports System.ComponentModel
Imports System.Reflection

Imports Model

Namespace Quote

    Public Class Detail
        Inherits Common.Detail

        Private _Properties As New Common.SaveableProperties

        Friend Sub New(ByVal header As Common.Header, _
                       ByVal product As Product)
            Me.QuoteHeader = header
            Me._Product = product
            Me._Quantity = 1
        End Sub

        <BrowsableAttribute(False)>
        Property QuoteHeader As Header

        <BrowsableAttribute(False)>
        Public Overrides ReadOnly Property QuoteDetailProperties As Object
            Get
                Return _Properties
            End Get
        End Property

        Public Sub SetProperties(ByVal props As Common.SaveableProperties)
            Me._Properties = props
        End Sub

        Public Overloads Property Qty() As Decimal
            Get
                Return Me._Quantity
            End Get

            Set(ByVal value As Decimal)
                If Not (value = _Quantity) Then
                    Me._Quantity = value
                    SendEvents()
                End If
            End Set
        End Property

    End Class

End Namespace

File: trunk/quote/Projects/PluginModel/quote/Detail.cs
namespace Model.Quote
{
    using System;
    using System.ComponentModel;
    using System.Linq;

    using Model.Common;

    public class Detail : Common.Detail
    {
        private Common.ISavableProperties properties = new QuoteSavableProperties();

        internal Detail(Common.Header header, Product product) : base(product, string.Empty, 1)
        {
            this.QuoteHeader = header as Quote.Header;
        }

        [BrowsableAttribute(false)]
        public Header QuoteHeader { get; set; }

        [BrowsableAttribute(false)]
        public override Model.Common.ISavableProperties QuoteDetailProperties 
        {
            get { return this.properties; }
        }

        public override decimal Qty 
        {
            get 
            { 
                return this.PrivateQty(); 
            }

            set 
            {
                if (!(value == this.PrivateQty())) 
                {
                    this.SetPrivateQty(value);
                    this.OnPropertyChanged();
                }
            }
        }
        
        public void SetProperties(Common.ISavableProperties props)
        {
            this.properties = props;
        }
    }
}